src/mixins/nr-affiliation-mixin.ts

```diff
@@ -136,7 +136,7 @@ export class NrAffiliationMixin extends Mixins(CommonMixin) {
   private getBusinessRequest (accountId: number, nr: NameRequestI): BusinessRequest {
     let name = 'registration'
     let legalType = nr.legalType
-    if (this.isSupportedIncorporationRegistration(nr.entity_type_cd)) {
+    if (this.isSupportedIncorporationRegistration(nr.entity_type_cd) && !this.isFirm) {
```

Comment:
The logic here was incorrect. Basically, we were always going into the if statement whether we're incorporating (BC, ULC etc...) or doing a registration (SPs, GPs, DBAs) since they're all in the FF.

Adding the `isFirm` prevents from going into the if statement when we're doing registration of SPs, GPs, or DBAs.

Reply:
Ah, OK... so if this is a firm then it keeps the name and legalType from lines 137-138 and then successfully creates a draft registration filing?

Man, this entire method could be cleaned up... too much back and forth logic! But it is Friday.

Reply:
Yup, you're precisely right Sev.

I'll have a look. See if I can clean it up in a nice way 🙌

Reply:
Sev, give me your thoughts on the code in the latest commit!

I've separated the logic into an if and else. That's easier to read and much cleaner I think.

Also, I noticed that the FF being used here is not needed. This is since we're controlling whether we're showing the button or not in the first place using the exact same FF.

Reply:
Oh yeah! Wayyyy better! 🦜

Reply:
I think you're right about the FF. Nice.

It was actually a bug before: if the FF was false then it would create a draft registration!

Looks fine to me.

Reply:
Awesome, thanks Sev!